spec/firebase_app_distribution_api_client_spec.rb

```diff
@@ -631,11 +631,255 @@
+    end
+  end
+
+  describe '#add_testers_to_group' do
+    let(:headers) do
+      {
+        'Authorization' => 'Bearer auth_token',
+        'Content-Type' => 'application/json',
+        'X-Client-Version' => "fastlane/#{Fastlane::FirebaseAppDistribution::VERSION}"
+      }
+    end
+    let(:project_number) { "project_number" }
+    let(:group_alias) { "group_alias" }
+
+    it 'is successful' do
+      emails = %w[tester1@test.com tester2@test.com]
+      payload = {
+        emails: emails,
+        createMissingTesters: true
+      }
+      stubs.post("/v1/projects/#{project_number}/groups/#{group_alias}:batchJoin", payload.to_json, headers) do |env|
+        [
+          200,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      result = api_client.add_testers_to_group(project_number, group_alias, emails)
+
+      expect(result.success?).to eq(true)
+    end
+
+    it 'fails and prints correct error message for a 400' do
+      emails = %w[tester1@test.com invalid_email_address]
+      payload = {
+        emails: emails,
+        createMissingTesters: true
+      }
+      stubs.post("/v1/projects/#{project_number}/groups/#{group_alias}:batchJoin", payload.to_json, headers) do |env|
+        [
+          400,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      expect { api_client.add_testers_to_group(project_number, group_alias, emails) }
+        .to raise_error(ErrorMessage::INVALID_EMAIL_ADDRESS)
+    end
+
+    it 'fails and prints correct error message for a 404' do
+      emails = %w[tester1@test.com tester2@test.com]
+      payload = {
+        emails: emails,
+        createMissingTesters: true
+      }
+      bad_project_number = "bad_project_number"
+      stubs.post("/v1/projects/#{bad_project_number}/groups/#{group_alias}:batchJoin", payload.to_json, headers) do |env|
+        [
+          404,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+      expect { api_client.add_testers_to_group(bad_project_number, group_alias, emails) }
+        .to raise_error(ErrorMessage::INVALID_TESTER_GROUP)
+    end
+  end
+
+  describe '#remove_testers_from_group' do
+    let(:headers) do
+      {
+        'Authorization' => 'Bearer auth_token',
+        'Content-Type' => 'application/json',
+        'X-Client-Version' => "fastlane/#{Fastlane::FirebaseAppDistribution::VERSION}"
+      }
+    end
+    let(:project_number) { "project_number" }
+    let(:group_alias) { "group_alias" }
+
+    it 'is successful' do
+      emails = %w[tester1@test.com tester2@test.com]
+      payload = { emails: emails }
+      stubs.post("/v1/projects/#{project_number}/groups/#{group_alias}:batchLeave", payload.to_json, headers) do |env|
+        [
+          200,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      result = api_client.remove_testers_from_group(project_number, group_alias, emails)
```

Comment:
Consider returning the body of the response, which would actually be null, and verifying that in the test instead of returning the response wrapper object.